lib/Conversion/HWToBTOR2/HWToBTOR2.cpp

```diff
@@ -458,21 +464,32 @@ struct ConvertHWToBTOR2Pass
     return width;
   }
 
+  // Emits the btor2 instructions required to define a state transition system
+  // for the given register
+  void finalizeGenRegVisit(Operation *reg, size_t width, Value next,
+                           Value resetVal) {}
+
   // Generates the transitions required to finalize the register to state
   // transition system conversion
   void finalizeRegVisit(Operation *op) {
-    // Check the register type (done to support non-firrtl registers as well)
-    auto reg = cast<seq::FirRegOp>(op);
+    int64_t width;
+    Value next, resetVal;
+
+    // Extract the operands depending on the register type
+    if (auto reg = dyn_cast<seq::CompRegOp>(op)) {
+      width = hw::getBitWidth(reg.getType());
+      next = reg.getInput();
+      resetVal = reg.getResetValue();
+    } else if (auto reg = dyn_cast<seq::FirRegOp>(op)) {
+      width = hw::getBitWidth(reg.getType());
+      next = reg.getNext();
+      resetVal = reg.getResetValue();
+    } else
+      op->emitError("Invalid register operation !");
```

Comment:
As I said, preferred LLVM style is to omit braces on single line bodys. That is, unless it harms readability. 😁
"For example, readability is also harmed if an if/else chain does not use braced bodies for either all or none of its members" (https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements)

More importantly though, this needs a `return`. Otherwise it continues with uninitialized variables.